### ISSUE TITLE
WiFi Guest Presence

### ISSUE THREAD

**Issue by Christian-CFP** [SeydX/homebridge-fritz-platform#215]
It would be nice if there were a presence sensor for the guest WiFi.
A global "Guest WiFi" Presence. Like the Anyone.

I know that I can make specific ones with IP or Mac but it would be nice if there is an global option to enable it so that I know there is someone logged in as guest in my WiFi.

**Comment by Christian-CFP** [comment]
Ziehst du das in Erwägung oder eher nicht?

**Comment by SeydX** [comment]
Ist in meiner Todo Liste

**Comment by SeydX** [comment]
Added in v6.0.16

**Comment by Christian-CFP** [comment]
Hey, doesnt seem to work. Logged me in but the sensor does not react to a login to the guest wifi.

**Comment by Christian-CFP** [comment]
Or is it like this because I am registered with an ip on the FB? Or that I am registered as a device in your plugin?

**Comment by Christian-CFP** [comment]
Nope tried it with a different phone. Sensor does not react.

**Comment by Christian-CFP** [comment]
Sorry, it works but it really does take a while until its reacting.

**Comment by SeydX** [comment]
Can u enable debug anf post the log pls

**Comment by Christian-CFP** [comment]
There is not much to see. But I guess I found the issue.
WPA3
If thats activated it takes so long or it does not recognize the guest sensor at all.

**Comment by Christian-CFP** [comment]
And it seems that it works just once... strange.
If I disconnected from the guest WiFi the sensor is off. If I reconnect the sensor stays off. Does not switch to on anymore.

**Comment by Christian-CFP** [comment]
Yes, I have to restart the FB (7.28) to set it to work. But that only works one time. If logged into the guest wifi the sensor is on. Disconnect from guest wifi, sensor off. Reconnect to guest wifi nothing, sensor still off.

**Comment by Christian-CFP** [comment]
Confirmed. It works just once. Than I have to restart the FB to get it back working (just once).

**Comment by SeydX** [comment]
hmm weird, i can not reproduce it, on my system it works perfect (7590)

```
[5.10.2021, 23:36:00] [FritzPlatform] Guest (motion): Someone connected to Guest Wifi
[5.10.2021, 23:36:00] [FritzPlatform] Seyit (motion): Occupancy state changed to NOT DETECTED
[5.10.2021, 23:36:00] [FritzPlatform] Seyit (motion): Wait 90s before switching state!
[5.10.2021, 23:36:20] [FritzPlatform] Guest (motion): No one connected to Guest Wifi
[5.10.2021, 23:36:32] [FritzPlatform] Seyit (motion): Occupancy state switched back to DETECTED
[5.10.2021, 23:36:40] [FritzPlatform] Guest (motion): Someone connected to Guest Wifi
[5.10.2021, 23:36:42] [FritzPlatform] Seyit (motion): Occupancy state changed to NOT DETECTED
[5.10.2021, 23:36:42] [FritzPlatform] Seyit (motion): Wait 90s before switching state!
[5.10.2021, 23:36:50] [FritzPlatform] Guest (motion): No one connected to Guest Wifi
[5.10.2021, 23:36:53] [FritzPlatform] Seyit (motion): Occupancy state switched back to DETECTED
```

**Comment by Christian-CFP** [comment]
That's strange. Even with 6.0.19:

`6.10.2021, 06:17:34 HOME Bridge FritzPlatform Fritz!Box 7590 (wifi_guest): ON
6.10.2021, 06:17:46 HOME Bridge FritzPlatform Guest (occupancy): Someone connected to Guest Wifi
6.10.2021, 06:17:59 HOME Bridge FritzPlatform Christian (occupancy): Occupancy state changed to NOT DETECTED
6.10.2021, 06:17:59 HOME Bridge FritzPlatform Christian (occupancy): Wait 30s before switching state!
6.10.2021, 06:18:09 HOME Bridge FritzPlatform Christian (occupancy): Occupancy state switched back to DETECTED
6.10.2021, 06:18:17 HOME Bridge FritzPlatform Guest (occupancy): No one connected to Guest Wifi
-- connected to Guest Wifi again (nothing)
6.10.2021, 06:23:02 HOME Bridge FritzPlatform Christian (occupancy): Occupancy state changed to NOT DETECTED
6.10.2021, 06:23:02 HOME Bridge FritzPlatform Christian (occupancy): Wait 30s before switching state!
6.10.2021, 06:23:33 HOME Bridge FritzPlatform Christian (occupancy): Occupancy state switched back to DETECTED
-- connected to Guest Wifi again (nothing)
6.10.2021, 06:24:04 HOME Bridge FritzPlatform Christian (occupancy): Occupancy state changed to NOT DETECTED
6.10.2021, 06:24:04 HOME Bridge FritzPlatform Christian (occupancy): Wait 30s before switching state!
6.10.2021, 06:24:45 HOME Bridge FritzPlatform Christian (occupancy): Occupancy state changed to DETECTED
6.10.2021, 06:24:45 HOME Bridge FritzPlatform Christian (occupancy): Wait 20s before switching state!`

**Comment by Christian-CFP** [comment]
Using also Repeater 1200 and 1750 as mesh. FW 7.27. 7590 is master.

**Comment by Christian-CFP** [comment]
I completely installed HOOBS again. Still the same. It has to do with the WiFi mesh from FB in combination with repeater 1200 and 1750 with the firmware.

FB 7490 mesh master: v7.28
1200 & 1750 mesh repeater: v7.27

**Comment by Christian-CFP** [comment]
FB 7490 mesh master: v7.29
1200 & 1750 mesh repeater: v7.27

Doesn't work at all. :(

**Comment by ShadowJonathan** [comment]
I've figured out that the bridge will not detect any guest wifi occupations if they're not connecting to the main router, if i connect anything to a repeater, it shows up in the Mesh overlay in the fritz box interface, but not in homekit/homebridge. If i connect to the main router, it does show up as occupied/motion-detected.

**Comment by krie** [comment]
I like this feature, but would it be possible to exclude some devices by MAC address from triggering the guest sensor?
In my case I have my work PCs in the guest network. For these I don’t want the guest sensor to trigger, because these are no ‚real‘ guests at my home.
Thank you!
